parsers/EIA.py

```diff
@@ -372,18 +373,24 @@ def fetch_consumption(
     session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
-):
+) -> List[Dict[str, Any]]:
     consumption = _fetch(
         zone_key,
         CONSUMPTION.format(REGIONS[zone_key]),
         session=session,
         target_datetime=target_datetime,
         logger=logger,
     )
+    batch = ConsumptionBatch(logger)
     for point in consumption:
-        point["consumption"] = point.pop("value")
+        batch.append(
+            zone_key=point["zoneKey"],
+            datetime=point["datetime"],
+            source=point["source"],
+            consumption=point["value"],
+        )
 
-    return consumption
+    return batch.to_dict()
```

Comment:
as mentioned yesterday, this function should be renamed as we expect to return a list of dict and not a dict. It is a bit confusing

Reply:
Done